services/core/PlatformDriverAgent/platform_driver/interfaces/home_assistant.py

```diff
@@ -0,0 +1,308 @@
+# implied, or assumes any legal liability or responsibility for the accuracy,
+# completeness, or usefulness or any information, apparatus, product,
+# software, or process disclosed, or represents that its use would not infringe
+# privately owned rights. Reference herein to any specific commercial product,
+# process, or service by trade name, trademark, manufacturer, or otherwise
+# does not necessarily constitute or imply its endorsement, recommendation, or
+# favoring by the United States Government or any agency thereof, or
+# Battelle Memorial Institute. The views and opinions of authors expressed
+# herein do not necessarily state or reflect those of the
+# United States Government or any agency thereof.
+#
+# PACIFIC NORTHWEST NATIONAL LABORATORY operated by
+# BATTELLE for the UNITED STATES DEPARTMENT OF ENERGY
+# under Contract DE-AC05-76RL01830
+# }}}
+
+
+import random
+from math import pi
+import json
+import sys
+from platform_driver.interfaces import BaseInterface, BaseRegister, BasicRevert
+from volttron.platform.agent import utils #added this to pull from config store 
+from volttron.platform.vip.agent import Agent
+import logging
+import requests
+from requests import get
+from collections.abc import MutableMapping
+
+_log = logging.getLogger(__name__)
+type_mapping = {"string": str,
+                "int": int,
+                "integer": int,
+                "float": float,
+                "bool": bool,
+                "boolean": bool}
+
+# def flatten_dict(d: MutableMapping, parent_key: str = '', sep: str ='.') -> MutableMapping:
+#     items = []
+#     for k, v in d.items():
+#         new_key = parent_key + sep + k if parent_key else k
+#         if isinstance(v, MutableMapping):
+#             items.extend(flatten_dict(v, new_key, sep=sep).items())
+#         else:
+#             items.append((new_key, v))
+#     return dict(items)
+
+class HomeAssistantRegister(BaseRegister):
+    def __init__(self, read_only, pointName, units, reg_type, attributes,
+                 default_value=None, description=''):
+        super(HomeAssistantRegister, self).__init__("byte", read_only, pointName, units,
+                                           description='')
+        self.reg_type = reg_type
+        self.attributes = attributes
+
+        if default_value is None:
+            self.value = self.reg_type(random.uniform(0, 100))
+        else:
+            try:
+                self.value = self.reg_type(default_value)
+            except ValueError:
+                self.value = self.reg_type()
+
+
+class Interface(BasicRevert, BaseInterface):
+    def __init__(self, **kwargs):
+        super(Interface, self).__init__(**kwargs)
+        self.point_name = None
+  
+    def configure(self, config_dict, registry_config_str): # grabbing from config
+        self.ip_address = config_dict.get("ip_address", "0.0.0.0")
+        self.access_token = config_dict.get("access_token", "access_token")
```

Comment:
Probably access_token should be None and raise an error if it's not found.

Reply:
same with volttron_topic, port, registry config.
isn't registry_config the same as registry_config_str just one is a dictionary and one is a string?